### ISSUE TITLE
Remove dependency on AspNetCore.All where it isn't needed

### ISSUE THREAD

**Issue by ptyang** [vany0114/EF.DbContextFactory#15]
It would be nice if AspNetCore.All could be removed from a dependency here, as it's quite large - and from looking at the extension/source file, there isn't a need there.

This would make using the EF.DbContextFactory nicer outside of the context of an ASP .NET Core application.

**Comment by vany0114** [comment]
Hi @ptyang 

You're right, you might want to use `EF.DbContextFactory` on a [Generic Host](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/generic-host?view=aspnetcore-2.2) or something like that.

BTW what is your use case? just curious...

So yeah, I'll let this issue open to work thru it in the near future, thanks!

**Comment by ptyang** [comment]
Actually I've just about wrapped up some changes in a repo where I've:
- Moved out the dependency on AspNetCore.All to the IntegrationTest where it's actually being used
- Upgraded both the EFCore.DbContextFactory.IntegrationTest project and the EFCore.DbContextFactory project to target .NET Core 2.1

Both projects build and tests still run fine, I'm going to have a PR for you if you want to take a look for ideas on upgrading the rest of the projects

**Comment by vany0114** [comment]
That sounds really good, go ahead with the PR!

**Comment by TAGC** [comment]
Yeah, I encountered an exception due to access by multiple threads to a shared DbContext and was looking into this project as a potential solution. However, I uninstalled when I saw it pull in the entire `Microsoft.AspNetCore.All` metapackage. For one thing that's a lot of dependencies but also it's pulling in the packages for ASP.NET Core 2 whereas I'm developing an ASP.NET Core 3 application.

So I'll hold off this library for now but probably take another look at it hen this and #19 are merged.
